vllm/executor/ray_gpu_executor.py

```diff
@@ -179,7 +188,7 @@ def sort_by_driver_then_worker_ip(worker):
             3. Finally, if the work is on a node with smaller IP address, it
                 should be placed first.
             """
-            ip = ray.get(worker.get_node_ip.remote())
+            ip = worker_to_ip[worker]
```

Comment:
@ruisearch42  this one looks concerning to me. we should change the tuple to sort, instead of using worker as the key. see the code from #11256 
<img width="607" alt="image" src="https://github.com/user-attachments/assets/8b56631c-cc06-4135-b6fa-31d8530c362e" />

Reply:
using an arbitrary python object as a key introduces quite unpredictable behavior and can have silent bugs.

Reply:
it's not about using an external dict, it's about using the worker object as a dict key, which implicitly calls its `__hash__` function.